contracts/crowdsale/price/IncreasingPriceCrowdsale.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.4.18;
+
+import "../validation/TimedCrowdsale.sol";
+import "../../math/SafeMath.sol";
+
+/**
+ * @title IncreasingPriceCrowdsale
+ * @dev Extension of Crowdsale contract that increases the price of tokens linearly in time. 
+ * Note that what should be provided to the constructor is the initial and final _rates_, that is,
+ * the amount of tokens per wei contributed. Thus, the initial rate must be greater than the final rate.
+ */
+contract IncreasingPriceCrowdsale is TimedCrowdsale {
+  using SafeMath for uint256;
+
+  uint256 public initialRate;
+  uint256 public finalRate;
+
+  /**
+   * @dev Constructor, takes intial and final rates of tokens received per wei contributed.
+   * @param _initialRate Number of tokens a buyer gets per wei at the start of the crowdsale
```

Comment:
Do we have to enforce that the linear function here must make the tokens increase in price? I expect a crowdsale that gets cheaper over time could be desired as well. I'm not sure there's a benefit to requiring an increase. We should let the creator make a linear function of their own slope. (and in that case this contract could be `LinearlyScaledRateCrowdsale`

Reply:
The point of restricting the price to be increasing was to be able to use unsigned ints throughout.. Plus I didn't see much use for a decreasing price crowdsale, but we could add this more general crowdsale flavor later if you think it might be useful!

Reply:
No worries, I'm happy with that logic 👍